### PR TITLE
Blog post for Finagle Block Party

### PR DESCRIPTION
**Pull request by kevinoliver**

### CODE REVIEW

**Comment by bryce-anderson** [comment]
Looks good!

File: source/blog/2016-09-01-block-party.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+[Await.result](http://twitter.github.io/util/docs/index.html#com.twitter.util.Await$@result[T](awaitable:com.twitter.util.Awaitable[T],timeout:com.twitter.util.Duration):T)
+or
+[Await.ready](http://twitter.github.io/util/docs/index.html#com.twitter.util.Await$@ready[T%3C:com.twitter.util.Awaitable[_]](awaitable:T):T).
+
+If your service is running Finagle 6.37, there is now a gauge exported
+as
+[scheduler/blocking\_ms](http://twitter.github.io/finagle/guide/Metrics#scheduler)
+which can be used to indentify how much time is being spent. You can
+verify that your service continues to operate correctly by checking
+the new lint rule added to TwitterServers
```
Comment (jcrossley) [inline source/blog/2016-09-01-block-party.md]: TwitterServer's

File: source/blog/2016-09-01-block-party.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+[/admin/lint](http://twitter.github.io/twitter-server/Admin.html#admin-lint)
+admin endpoint.
+
+Implicit blocking happens when your code runs a potentially expensive
+operation, for example making network calls on the event loop thread
+using an synchronous API. Unfortunately, these calls are difficult to
```
Comment (jcrossley) [inline source/blog/2016-09-01-block-party.md]: a synchronous API

File: source/blog/2016-09-01-block-party.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+track down and we do not yet have any tooling to help on this front.
+
+✮ *Bummer, it is blocking. How do I find out what’s blocking?*
+
+You can do a deploy with an extra system property set that will log
+the stacktraces for a fraction of the blocking calls. Given that the
+code is about to block, the extra overhead of logging the stacktrace
+shouldnt be a significant overhead. However, if your service has a
```
Comment (jcrossley) [inline source/blog/2016-09-01-block-party.md]: shouldn't

File: source/blog/2016-09-01-block-party.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+blocking is when code synchronously waits on a Future from an event
+loop thread. In Finagle, this is done by making calls to
+[Await.result](http://twitter.github.io/util/docs/index.html#com.twitter.util.Await$@result[T](awaitable:com.twitter.util.Awaitable[T],timeout:com.twitter.util.Duration):T)
+or
+[Await.ready](http://twitter.github.io/util/docs/index.html#com.twitter.util.Await$@ready[T%3C:com.twitter.util.Awaitable[_]](awaitable:T):T).
+
+If your service is running Finagle 6.37, there is now a gauge exported
+as
+[scheduler/blocking\_ms](http://twitter.github.io/finagle/guide/Metrics#scheduler)
+which can be used to indentify how much time is being spent. You can
```
Comment (ryanoneill) [inline source/blog/2016-09-01-block-party.md]: identify

File: source/blog/2016-09-01-block-party.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+You can do a deploy with an extra system property set that will log
+the stacktraces for a fraction of the blocking calls. Given that the
+code is about to block, the extra overhead of logging the stacktrace
+shouldn't be a significant overhead. However, if your service has a
+large amount of blocking, you may want to limit this fraction to avoid
+filling up your logs. You can set it via
+`-Dcom.twitter.concurrent.schedulerSampleBlockingFraction=$fraction`
+where `$fraction` must be between `0.0` and `1.0`, inclusive.
+
+This will output log that should point you to the code doing the
```
Comment (ryanoneill) [inline source/blog/2016-09-01-block-party.md]: 'a log' or 'log messages'

**Pull request merged**